clients/src/main/java/org/apache/kafka/common/network/SslChannelBuilder.java

```diff
@@ -45,12 +46,13 @@ public void configure(Map<String, ?> configs) throws KafkaException {
         }
     }
 
-    public KafkaChannel buildChannel(String id, SelectionKey key, int maxReceiveSize) throws KafkaException {
+    @Override
+    public KafkaChannel buildChannel(String id, SelectionKey key, int maxReceiveSize, MemoryPool memoryPool) throws KafkaException {
         try {
             SslTransportLayer transportLayer = buildTransportLayer(sslFactory, id, key);
             Authenticator authenticator = new DefaultAuthenticator();
             authenticator.configure(transportLayer, this.principalBuilder, this.configs);
-            return new KafkaChannel(id, transportLayer, authenticator, maxReceiveSize);
+            return new KafkaChannel(id, transportLayer, authenticator, maxReceiveSize, memoryPool != null ? memoryPool : MemoryPool.NONE);
```

Comment:
i was trying to be safe, so i "support" nulls by translating them to NONE. if you want me to choose either null or NONE (instead of both) - just choose which.